### PR TITLE
Update various things in the build image

### PR DESCRIPTION
**Pull request by sylr**
**What this PR does**:

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [ ] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by bboreham** [comment]
I changed the title since we have a few "Dockerfile"s

File: build-image/Dockerfile [hunk 1]
```diff
@@ -1,8 +1,8 @@
-FROM golang:1.13.3-stretch
+FROM golang:1.13-buster
```
Comment (pracucci) [inline build-image/Dockerfile]: I would suggest to specify the golang minor version to have reproducible builds. We can upgrade to the latest `1.13.6` tho.

**Comment by pracucci** [comment]
For the records, the new `build-image` (`dockerfile-3860bbe9-WIP`) is 32MB bigger then the current one (`validate-k8s-specs-7c217ee7`). Not dramatic, but worth to note.

Screen Shot 2020-01-28 at 16 35 09

**Comment by pracucci** [comment]
> I've published for your the Docker image quay.io/cortexproject/build-image:dockerfile-3860bbe9-WIP [...] Can you [...] replace the image URL in .circleci/config.yml and push it, so that we can test if the CI pass?

May you also address this, please?

**Comment by sylr** [comment]
@pracucci do you know what should be done to pass CI ?

**Comment by pracucci** [comment]
_Let me try to push changes directly to this PR, to contribute to make this PR into a mergeable state._

**Comment by gouthamve** [comment]
Superseded by https://github.com/cortexproject/cortex/pull/2480

**Pull request closed without merging**